### PR TITLE
Add a settings screen to pick the opam binary

### LINKED ISSUE #5: Provide a preference screen to pick opam path
The plugin only works if opam is in the path. We should allow use to choose between the default opam in the path and a configured opam location.
We should ensure all the configuration dependent on opamCommand are reloaded when we change the opam path.

### PR DESCRIPTION
**Pull request by beajeanm**
Should fix #5

### CODE REVIEW

File: src/test/kotlin/org/hoggar/test/OcamlC.kt [hunk 1]
```diff
@@ -18,7 +15,7 @@ object OcamlC {
         val fo = createTempFile()
 
         try {
-            val pb = opamCommand.processBuilder("ocamlc", "-dparsetree", src.absolutePath)
+            val pb = OpamCommand.processBuilder("ocamlc", "-dparsetree", src.absolutePath)
```
Comment (tartur) [inline src/test/kotlin/org/hoggar/test/OcamlC.kt]: minor: ideally all opam commands should be encapsulated in OpamCommand methods

File: src/main/kotlin/org/hoggar/merlin/OpamCommand.kt [hunk 2]
```diff
@@ -1,42 +1,64 @@
...
     }
 
-    private fun makeCommandString(vararg parameters: String): String {
-        val userHome = System.getProperty("user.home")
-        val cmd = """
-            . $userHome/.opam/opam-init/init.zsh > /dev/null 2> /dev/null || true
-            cd ~/Code/ocaml-test; ${parameters.joinToString(" ")}
-            """
-        return cmd
+    private fun binFolder(): String {
```
Comment (tartur) [inline src/main/kotlin/org/hoggar/merlin/OpamCommand.kt]: when is this called? is it a Kotlin feature?

File: src/main/kotlin/org/hoggar/merlin/OpamCommand.kt [hunk 3]
```diff
@@ -1,42 +1,64 @@
...
-        return output.split("\n")
-                .filter { it.startsWith("bin") }
-                .map { it.split(Regex("\\s+"))[1] }
-                .first()
+
+        private fun makeCommandString(vararg parameters: String): String {
+            val userHome = System.getProperty("user.home")
+            return """
+            . $userHome/.opam/opam-init/init.zsh > /dev/null 2> /dev/null || true
+            cd ~/Code/ocaml-test; ${parameters.joinToString(" ")}
```
Comment (tartur) [inline src/main/kotlin/org/hoggar/merlin/OpamCommand.kt]: is ~/Code/ocaml-test a common folder?!

File: src/main/kotlin/org/hoggar/merlin/OpamCommand.kt (hunk 2 above)
Comment (beajeanm) [reply src/main/kotlin/org/hoggar/merlin/OpamCommand.kt]: What? The bin folder function? 
It's just a regular private function, called by reload path if the current binary is different from the one used for the previous call.

File: src/test/kotlin/org/hoggar/test/OcamlC.kt (hunk 1 above)
Comment (beajeanm) [reply src/test/kotlin/org/hoggar/test/OcamlC.kt]: Not sure what you mean, ocamlc is not an opam command, we used opam to find the right binary.
Anyway, I haven't paid much attention to this test, it comes from the old code. I just fixed the compilation.

File: src/main/kotlin/org/hoggar/merlin/OpamCommand.kt (hunk 2 above)
Comment (tartur) [reply src/main/kotlin/org/hoggar/merlin/OpamCommand.kt]: Sorry didn't caught it!

**Pull request merged**